src/odemis/gui/comp/text.py

```diff
@@ -778,6 +810,33 @@ def on_char(self, evt):
             self._display_pretty()  # Update the GUI immediately
             self._send_change_event()
 
+    def get_log_step(self, value, k):
+        """
+        Calculates the step size by which the current value should be increased/decreased.
+        :parameter value: (float) Current value displayed in widget and that should be increased/decreased
+        :parameter k: (float) Order by which the value should be increased/decreases.
+            Is dependent on key combination pressed.
+        :return: (float) Step by which the current value should be increased/decreased (already contains the sign).
+        """
+        if k < 0:  # arrow done was pressed -> decrease of value requested
+            # Up/down keys are not just "opposite" (down = -up). They must compensate each other so that in
+            # (almost) all cases pressing up then down (or down then up) returns to the original value.
+            # If down was just "-up", this wouldn't work on the magnitude transitions. For example 9->10.
+            # So, instead, we compute the "retraction" of the "up function" (which is injective).
+            # In practice, this results in computing the magnitude of a "little bit" smaller value.
+            value *= (1 + k / 10)
+        try:
+            magnitude = int(math.floor(math.log10(abs(value))))
+            if magnitude <= -15:
+                raise ValueError("Value is so small, so set it zero.")
+        except ValueError:
+            return math.copysign(self.key_step_min, k)
+        step = (10 ** magnitude) * k
+        if self._num_type is int and abs(step) < 1:  # if integer, change at least by 1
```

Comment:
Could we just write something like this?
```
   if self.key_step_min and abs(step) < self.key_step_min:
        step = math.copysign(self.key_step_min, step)
```
Then, num_type can be dropped, and we just pass `key_step_min=1` by default on the *IntText classes?

Reply:
I think that code would then overwrite all steps that are smaller then key_step_min though it is possible to do smaller steps. Key_step_min is calculated based on the min/max values, so for the trigger delay it was always 1e-5. However, we would like to allow steps smaller than that. Or do I miss something?
I agree the key_step_min + num_type is not really nice yet...